src/main/java/org/apache/commons/io/filefilter/AbstractFileFilter.java

```diff
@@ -37,7 +37,13 @@
      * @return true if this file matches the test
      */
     @Override
+    @SuppressWarnings("nullness:argument.type.incompatible")
     public boolean accept(final File file) {
+        // File.getParentFile() can return null, and null is not documented
+        // as a permitted file argument for FilenameFilter.accept(File,String).
+
+        // However following code does not lead to any Exception at run time
```

Comment:
The documentation says that a client should override *at least one* of the accept methods.
You are saying that the documentation is incorrect.
You are reporting  that if this method is not overridden but the other one is, there is an infinite loop.
You should report this bug to the upstream project.